src/main/scala/org/scalajs/tools/tsimporter/Importer.scala

```diff
@@ -280,6 +280,14 @@ class Importer(val output: java.io.PrintWriter) {
         TypeRef.Singleton(QualifiedName((expr.qualifier :+ expr.name).map(
             ident => Name(ident.name)): _*))
 
+      case TupleType(targs) =>
+        if (targs.forall( _ == targs.head)) {
+          TypeRef(QualifiedName.Array, List(typeToScala(targs.head)))
```

Comment:
covariant would be better, needs more refactoring ( or a hint: maybe extend TypeRef)